docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,235 @@
+  - extensions
+  - ui-extensions
+sidebar_position: 20
+---
+
+An "extension" is either a link or a React component defined by a plugin and rendered either somewhere in the core Grafana UI or in an other app plugin.
+[Read more about extensions under key concepts](../../key-concepts/ui-extensions.md).
+
+| Type          | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Link**      | Links have a `path` and an `onClick()` property. <br /><br /> **When to use?** <br /> Use links if you would like to give plugins a way to define custom user actions for a part of your UI. These actions can either just be cross-links to the plugin, or using `onClick()` methods they can implement a more interactive on-page experience with a modal. <br /><br /> **API reference** <br /> - [`.addLink()`](../../reference/ui-extensions/addLink.md) - registering a link from a plugin <br /> - [`usePluginLinks()`](../../reference/ui-extensions/usePluginLinks.md) - fetching links registered for an extension point |
+| **Component** | Components are React components that can be used to render a custom user experience. <br /><br /> **When to use?** <br /> Use components if you would like to give more freedom for plugins to extend your UI, for example to extend a configuration form with custom parts. <br /><br /> **API reference** <br /> - [`.addComponent()`](../../reference/ui-extensions/addComponent.md) - registering a component from a plugin <br /> - [`usePluginComponents()`](../../reference/ui-extensions/usePluginComponents.md) - fetching components registered for an extension point                                                   |
+
+## Links
+
+### Best practices for adding links
+
+- **Define a `path`** - it is always good if your action can also be opened in a new tab using a link.
```

Comment:
I don't think we can claim this as a best practice as it basically says "dont do onClick"

Reply:
nit: elsewhere with best practices you've used new lines rather than - , we should be consistent

Reply:
Hmm, yeah, maybe you are right 🤔 

What do you think @sunker / @mckn ?

Reply:
hmm, I guess it depends on. If the action will trigger a modal then a path isn't applicable so I guess it depends on the flow. Maybe rephrase it to something like:

When the link results in a navigation favor `path` over `onClick` to create native links for the browser.

Reply:
This feels like a general guidance in web development, so not sure we need to describe this here. Me and @mckn agreed to removed this bullet point.